build.sh

```diff
@@ -94,8 +94,18 @@ else
 	unset BUILD_PARAMS
 fi
 echo ${BUILD_PARAMS}
-docker build ${BUILD_PARAMS} --build-arg AUTOTUNE_VERSION=${DOCKER_TAG} -t ${AUTOTUNE_DOCKER_IMAGE} -f ${AUTOTUNE_DOCKERFILE} .
-check_err "Docker build of ${AUTOTUNE_DOCKER_IMAGE} failed."
-docker build ${BUILD_PARAMS} --build-arg AUTOTUNE_VERSION=${DOCKER_TAG} -t ${OPTUNA_DOCKER_IMAGE} -f ${OPTUNA_DOCKERFILE} .
-check_err "Docker build of ${OPTUNA_DOCKER_IMAGE} failed."
+
+docker build ${BUILD_PARAMS} --format=docker --build-arg AUTOTUNE_VERSION=${DOCKER_TAG} -t ${AUTOTUNE_DOCKER_IMAGE} -f ${AUTOTUNE_DOCKERFILE} .
+err=$?
+if [ ${err} -ne 0 ]
+then
+  docker build ${BUILD_PARAMS} --build-arg AUTOTUNE_VERSION=${DOCKER_TAG} -t ${AUTOTUNE_DOCKER_IMAGE} -f ${AUTOTUNE_DOCKERFILE} .
+  check_err "Docker build of ${AUTOTUNE_DOCKER_IMAGE} failed."
+  docker build ${BUILD_PARAMS} --build-arg AUTOTUNE_VERSION=${DOCKER_TAG} -t ${OPTUNA_DOCKER_IMAGE} -f ${OPTUNA_DOCKERFILE} .
+  check_err "Docker build of ${OPTUNA_DOCKER_IMAGE} failed."
+else
+  docker build ${BUILD_PARAMS} --format=docker --build-arg AUTOTUNE_VERSION=${DOCKER_TAG} -t ${OPTUNA_DOCKER_IMAGE} -f ${OPTUNA_DOCKERFILE} .
+  check_err "Docker build of ${OPTUNA_DOCKER_IMAGE} failed."
+fi
```

Comment:
1. Please redirect the output of the first build command to a variable and make sure no error messages are seen on the console.
2. Check for the error message to determine the next course of action
```
docker build ${BUILD_PARAMS} --format=docker --build-arg AUTOTUNE_VERSION=${DOCKER_TAG} -t ${AUTOTUNE_DOCKER_IMAGE} -f ${AUTOTUNE_DOCKERFILE} . 2>/tmp/hpo_build_log.$$
build_error=$(grep "Error:" /tmp/hp_build_log.$$)
if [ -n "${build_error}" ]; then
    ...
else
    ...
fi
```
3. In our code we use the following syntax
    ```
         if  [cond]; then
         fi
    ```
All scripts use tabs and not spaces

Reply:
Done